crates/doc/src/inference.rs

```diff
@@ -1489,6 +1489,13 @@ impl Shape {
             );
         };
     }
+
+    pub fn to_serde(self) -> Result<Value, serde_json::Error> {
```

Comment:
I'd prefer we don't add this to the public interface.
Callers can do `serde_json::to_value(&SchemaBuilder::new(shape).root_schema())` for the time being.

I'd like to do further refactors of the `SchemaBuilder` stuff.

Reply:
kk. I was trying to follow [`HeapNode::from_serde()`](https://github.com/estuary/flow/blob/master/crates/doc/src/heap_de.rs#L18-L26), but also happy to not add to the public interface 👍

Reply:
done!